edb/server/bootstrap.py

```diff
@@ -1828,27 +1843,36 @@ async def _bootstrap(ctx: BootstrapContext) -> None:
     else:
         superuser_uid = uuidgen.uuid1mc()
 
-    in_dev_mode = devmode.is_in_dev_mode()
-    # Protect against multiple EdgeDB tenants from trying to bootstrap
-    # on the same cluster in devmode, as that is both a waste of resources
-    # and might result in broken stdlib cache.
-    if in_dev_mode:
-        await ctx.conn.sql_execute(b"SELECT pg_advisory_lock(3987734529)")
-
     if backend_params.has_create_database:
         new_template_db_id = await _create_edgedb_template_database(ctx)
         tpl_db = cluster.get_db_name(edbdef.EDGEDB_TEMPLATE_DB)
         conn = await cluster.connect(database=tpl_db)
+
+        # HACK: When bootstrapping against managed Digital Ocean
+        # databases, the database occasionally had a server process
+        # crash if we kept our main connection open while initializing
+        # the template DB (!).
+        # So we close the connection now, and we'll reopen it once
+        # we've finished with the template DB.
+        ctx.conn.terminate()
+        ctx.conn = None  # type: ignore
     else:
         new_template_db_id = uuidgen.uuid1mc()
 
-    try:
-        if backend_params.has_create_database:
-            tpl_ctx = ctx._replace(conn=conn)
-            conn.add_log_listener(_pg_log_listener)
-        else:
-            tpl_ctx = ctx
+    if backend_params.has_create_database:
```

Comment:
(trivial) this `if ... else ...` arms can be merged into the previous `if ... else ...` with identical condition now